k8s-plugin/src/main/kotlin/com/amazon/spinnaker/keel/k8s/resolver/KustomizeResourceHandler.kt

```diff
@@ -1,66 +1,134 @@
 package com.amazon.spinnaker.keel.k8s.resolver
 
 import com.amazon.spinnaker.keel.k8s.*
+import com.amazon.spinnaker.keel.k8s.exception.InvalidArtifact
 import com.amazon.spinnaker.keel.k8s.exception.MisconfiguredObjectException
-import com.amazon.spinnaker.keel.k8s.model.KustomizeResourceSpec
+import com.amazon.spinnaker.keel.k8s.exception.NoVersionAvailable
+import com.amazon.spinnaker.keel.k8s.model.GitRepoArtifact
 import com.amazon.spinnaker.keel.k8s.model.K8sObjectManifest
+import com.amazon.spinnaker.keel.k8s.model.KustomizeResourceSpec
+import com.amazon.spinnaker.keel.k8s.resolver.FluxManifestUtil.generateGitRepoManifest
 import com.amazon.spinnaker.keel.k8s.service.CloudDriverK8sService
-import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
-import com.fasterxml.jackson.module.kotlin.readValue
+import com.netflix.spinnaker.keel.api.DeliveryConfig
 import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ResourceDiff
+import com.netflix.spinnaker.keel.api.actuation.Task
 import com.netflix.spinnaker.keel.api.actuation.TaskLauncher
 import com.netflix.spinnaker.keel.api.plugins.Resolver
 import com.netflix.spinnaker.keel.api.support.EventPublisher
 import com.netflix.spinnaker.keel.orca.OrcaService
+import com.netflix.spinnaker.keel.persistence.KeelRepository
+import com.netflix.spinnaker.keel.persistence.NoMatchingArtifactException
+import kotlinx.coroutines.coroutineScope
 
 class KustomizeResourceHandler(
     override val cloudDriverK8sService: CloudDriverK8sService,
     override val taskLauncher: TaskLauncher,
     override val eventPublisher: EventPublisher,
     orcaService: OrcaService,
-    override val resolvers: List<Resolver<*>>
+    override val resolvers: List<Resolver<*>>,
+    val repository: KeelRepository
 ) : GenericK8sResourceHandler<KustomizeResourceSpec, K8sObjectManifest>(
     cloudDriverK8sService, taskLauncher, eventPublisher, orcaService, resolvers
 ) {
     override val supportedKind = KUSTOMIZE_RESOURCE_SPEC_V1
 
     public override suspend fun toResolvedType(resource: Resource<KustomizeResourceSpec>): K8sObjectManifest {
-
-        if (resource.spec.template.apiVersion == FLUX_KUSTOMIZE_API_VERSION && resource.spec.template.kind == FLUX_KUSTOMIZE_KIND) {
-            return super.toResolvedType(resource)
+        log.debug("attempting to resolve resource for git")
+        if (resource.spec.template.kind != FLUX_KUSTOMIZE_KIND || resource.spec.template.apiVersion != FLUX_KUSTOMIZE_API_VERSION) {
+            throw MisconfiguredObjectException(
+                "incorrect kind or api version supplied. supplied: ${resource.spec.template.kind} + ${resource.spec.template.apiVersion}" +
+                        " supported: $FLUX_KUSTOMIZE_KIND + $FLUX_KUSTOMIZE_API_VERSION"
+            )
```

Comment:
can we also separate it into a `verifyKustomizeResource` similar to what we have done for HELM? looks cleaner.